DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/use_library.cpp

```diff
@@ -0,0 +1,107 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "lib_rtl.hpp"
+#include "exception_handler.hpp"
+
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/experimental/pipe_properties.hpp>
+#include <sycl/ext/intel/experimental/pipes.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
```

Comment:
review kevin's interfaces overview; the headers should be different
```suggestion
#include <sycl/ext/intel/ac_types/ac_int.hpp>
// oneAPI headers
#include <sycl/ext/intel/fpga_extensions.hpp>
#include <sycl/ext/intel/experimental/fpga_kernel_properties.hpp>
#include <sycl/ext/oneapi/annotated_arg/annotated_arg.hpp>
#include <sycl/sycl.hpp>

#include "lib_rtl.hpp"

#include "exception_handler.hpp"
```

Reply:
Should not the sycl.hpp be on top to avoid clash?
Is it a must now that we need to use annotated_arg for the invocation interface?

Reply:
that's fixed in 2024.0

`annotated_arg` is for kernel *arguments*, so pointers and scalar types that are just passed to the kernel. This design uses pipes for input/output so `annotated_arg` should not be needed.